### PR TITLE
Fix consignment status endpoint bugs & Refactor code a little

### PR DESCRIPTION
**Pull request by techncl**

### CODE REVIEW

File: src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/ConsignmentStatusRepositorySpec.scala [hunk 1]
```diff
@@ -41,9 +43,59 @@ class ConsignmentStatusRepositorySpec extends AnyFlatSpec with TestDatabase with
...
 
-    consignmentUploadStatus should be(empty)
+    consignmentStatuses.length should be(3)
+
+    consignmentStatuses.head.statustype should be(statusTypeOne)
+    consignmentStatuses(1).statustype should be(statusTypeTwo)
+    consignmentStatuses(2).statustype should be(statusTypeThree)
+  }
+
+  "getConsignmentStatus" should "return only the consignment status for the consignment specified" in {
```
Comment (suzannehamilton) [inline src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/ConsignmentStatusRepositorySpec.scala:77]: Thanks for adding these extra edge cases. 👍

File: src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/ConsignmentStatusRepositorySpec.scala [hunk 2]
```diff
@@ -68,4 +120,44 @@ class ConsignmentStatusRepositorySpec extends AnyFlatSpec with TestDatabase with
...
+  "updateConsignmentStatus" should "only update the value of the status type passed in" in {
+    val db = DbConnection.db
+    val consignmentStatusRepository = new ConsignmentStatusRepository(db)
+    val consignmentId = UUID.fromString("2e998acd-6e87-4437-92a4-e4267194fe38")
+    val userId = UUID.fromString("7f7be445-9879-4514-8a3e-523cb9d9a188")
+    val statusTypeOne = "TransferAgreement"
+    val statusTypeTwo = "Upload"
+    val statusTypeThree = "Export"
+    val statusValueOne = "Completed"
+    val statusValueTwo = "Completed"
```
Comment (suzannehamilton) [inline src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/ConsignmentStatusRepositorySpec.scala:133]: Should statues one and two be something other than Completed? Otherwise this test can't tell whether the update method has changed them all to "Completed".

Comment (techncl) [reply src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/ConsignmentStatusRepositorySpec.scala:133]: Ah yes! Thanks, I will change this

**Review by suzannehamilton — APPROVED** [review]
